### DISCUSSION TITLE
Error 0x80070057: The parameter is incorrect" when extracting file

### DISCUSSION THREAD

**Discussion by parveenvayu** [gildas-lormeau/zip.js#543] [Q&A] [answered]
Error 0x80070057: The parameter is incorrect" when extracting file

 const zipBlob = await (async () => {
     //zip.configure({ chunkSize: 128, useWebWorkers: false });
     let blobWriter = new BlobWriter("application/zip");
     let zipWriter = new ZipWriter(blobWriter, { password: "password", zipCrypto: true });
     await zipWriter.add(`Vayu_AllRequests_${exportFromDate}_${exportToDate}.csv`, new BlobReader(csvBlob));
     await zipWriter.close();
     return blobWriter.getData();
 })();

 // Trigger file download
 saveAs(zipBlob, `Vayu_AllRequests_${exportFromDate}_${exportToDate}.zip`);

**Comment by gildas-lormeau** [comment, marked as answer]
There is probably invalid characters in the filename, see https://stackoverflow.com/questions/57165960/error-0x80070057-the-parameter-is-incorrect-when-unzipping-files.

**Reply by parveenvayu** [reply]
Yes it worked after filename changing without reserved characters
